Document the group resource configurations (#8893)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/groups.md

  ```

### Group properties

The following properties are available when defining a group:

| Property | Required | Description |
|----------|----------|-------------|
| `name` | Required | A unique name for the group within the project. Used to assign resources via the [`group` config](/reference/resource-configs/group). |

> | `owner` | Required | Identifies who is responsible for the group. Must include either `name` or `email`. |

Author: The name is arbitrary. It would be the field they provide similar to other implementations of `owner`

> | `owner` | Required | Identifies who is responsible for the group. Must include either `name` or `email`. |

Author: ```suggestion
| `owner` | Required | Identifies who is responsible for the group. Must include either `name:` or `email:`. |
```

> | `description` | Optional | <VersionBlock firstVersion="1.10">A human readable description of the group's purpose. Supports markdown and the [`doc` Jinja function](/reference/dbt-jinja-functions/doc).</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |

Reviewer: ```suggestion
| `description` | Optional | A human readable description of the group's purpose. Supports markdown and the [`doc` Jinja function](/reference/dbt-jinja-functions/doc).  Supported in v1.10 and later. |
```

  ```

### Group properties

The following properties are available when defining a group:

| Property | Required | Description |
|----------|----------|-------------|
| `name` | Required | A unique name for the group within the project. Used to assign resources via the [`group` config](/reference/resource-configs/group). |

> | `owner` | Required | Identifies who is responsible for the group. Must include either `name` or `email`. |

Reviewer: Where is the name from? Their user name?

> | `description` | Optional | <VersionBlock firstVersion="1.10">A human-readable description of the group's purpose. Supports markdown and the [`doc` Jinja function](/reference/dbt-jinja-functions/doc).</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |

Author: ```suggestion
| `description` | Optional | <VersionBlock firstVersion="1.10">A human readable description of the group's purpose. Supports markdown and the [`doc` Jinja function](/reference/dbt-jinja-functions/doc).</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |
```

> | `config.meta` | Optional | <VersionBlock firstVersion="1.10">A dictionary of arbitrary key-value metadata about the group. Useful for storing information like cost center, data classification, or team contact details.</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |

Author: ```suggestion
| `config.meta` | Optional | <VersionBlock firstVersion="1.10">A dictionary of arbitrary key/value metadata about the group. Useful for storing information like cost center, data classification, or team contact details.</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |
```

> | `config.meta` | Optional | <VersionBlock firstVersion="1.10">A dictionary of arbitrary key/value metadata about the group. Useful for storing information like cost center, data classification, or team contact details.</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |

Reviewer: Do we need to do this versioning? We don't show 1.9 in the menu. Maybe we can just have it say not supported in v1.9 an dearlier?

Author: Oop! Forgot we removed it

> | `config.meta` | Optional | <VersionBlock firstVersion="1.10">A dictionary of arbitrary key/value metadata about the group. Useful for storing information like cost center, data classification, or team contact details.</VersionBlock><VersionBlock lastVersion="1.9">Not supported in this version.</VersionBlock> |

Author: ```suggestion
| `config.meta` | Optional | A dictionary of arbitrary key/value metadata about the group. Useful for storing information such as cost centers, data classifications, or team contact details.
```

> <VersionBlock firstVersion="1.10">

Reviewer: Same version note

> <VersionBlock firstVersion="1.10">

Author: ```suggestion
```

<File name='models/marts/finance/finance.yml'>

```yaml
groups:
  - name: finance

>     description: "All finance-related models owned by the Finance team."

Author: ```suggestion
    description: "All models owned by the Finance team."
```

    owner:
      email: finance@jaffleshop.com
    config:
      meta:
        data_owner: Finance team
        cost_center: finance
        data_classification: sensitive
```

</File>

> </VersionBlock>

Author: ```suggestion
```
